src/components/LHNOptionsList/OptionRowLHNData.js

```diff
@@ -88,6 +92,14 @@ function OptionRowLHNData({
     const parentReportAction = parentReportActions[fullReport.parentReportActionID];
 
     const optionItemRef = useRef();
+
+    const lastTransaction = useMemo(() => {
+        const sortedReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(reportActions);
```

Comment:
Question: could we instead do this logic in [SidebarUtils.getOptionData](https://github.com/Expensify/App/blob/8116a8e8924dd96ca58e780647524801e9cbe2e8/src/libs/SidebarUtils.js#L182)? We already have `lastReportActions` defined there and we could only add a `linkedTransaction` key to the `result` object. I think that'd help with performance given that we wouldn't need to sort reportActions again

Reply:
We need to do this logic to decide when to rerender the `OptionRowLHNData`, so it has to be out of [SidebarUtils.getOptionData](https://github.com/Expensify/App/blob/8116a8e8924dd96ca58e780647524801e9cbe2e8/src/libs/SidebarUtils.js#L182) and in the OptionRowLHNData right?

Reply:
maybe I'm missing something, but we pass `optionItem` as a [prop](https://github.com/Expensify/App/blob/5b2c016390096a9c27c929daaf1c7921b11c6742/src/components/LHNOptionsList/OptionRowLHNData.js#L116), so if [getOptionData](https://github.com/Expensify/App/blob/5b2c016390096a9c27c929daaf1c7921b11c6742/src/components/LHNOptionsList/OptionRowLHNData.js#L93) returned a different value we'd rerender, no?

Reply:
`lastTransaction` is just a mere dependency for `optionItem` to recompute. So you need it in the first place to decide when to rerender

Reply:
Sounds good. We can keep it as is then. Thanks for clarifying!